aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -1366,7 +1366,7 @@ impl AptosVM {
                 }
 
                 let size_if_old_module_exists = module_storage
-                    .fetch_module_size_in_bytes(addr, name)?
+                    .unmetered_get_module_size(addr, name)?
                     .map(|v| v as u64);
                 if let Some(old_size) = size_if_old_module_exists {
```

Comment:
I see, so it is always unmetered, that's what the name implies, and here for example is a case where we actually charge the gas based on the read.
Would it make sense to have get_module_size take an Optional gas meter, and do below logic when it's some, and otherwise be unmetered?

Reply:
Yes! In general, the idea is that:
1. `unmetered_..` is not safe to use, so one should use `Loader` interfaces which have gas meter and context for traversal.
2. unmetered APIs is used to implement charging logic in `Loader` (so it is safe and technically unavoidable). This is to avoid propagating metering logic deep into Block-STM which implements `ModuleStorage`

Reply:
`get_module_size` does not need gas meter because it is used for metering, but what can happen is
`get_module` taking this optional gas meter (or even gas meter, because we can always pass `UnmeteredGasMeter) but then every implementation of `get_module` need to provide its own charging logic, i.e. block-stm, adapter over state view, etc. I therefore decided to have a VM-only `Loader` that does charging, but uses raw storage API in a safe metered manner. 

Ideally, all usages of `impl ModuleStorage` should be replaced by `impl Loader`: I eventually did it for Move VM (see next PRs) but for Aptos VM things are more difficult, so loader is created only when you actually call into Move code (we can probably move it more up I think)